### PR TITLE
add sort by filter

### PR DESCRIPTION
**Pull request by connorwriter**

### CODE REVIEW

**Comment by Harrison365** [comment]
This is looking good 👍 Just a couple of things needed:
 
- Need order_by ascending or descending choice. 
- Will need default sort by and order by (currently loads in random order)

Thanks Connor

**Comment by NRMSMITH** [comment]
- [ ] your url should change when the sort by and order change - meaning that I can send it to my friends in order to share a specific group of reviews!

Please resubmit once done!

**Comment by NRMSMITH** [comment]
The value is not coming down onto the page to show the rendering and order that you want.
Also, in your URL, you have 'category' listed but no value assigned?

Have a look again at the relationship between your URL and the values rendered.

File: src/api.js [hunk 1]
```diff
@@ -4,13 +4,25 @@ const ncGamesApi = axios.create({
...
-    return ncGamesApi.get(`/reviews?category=${category}`);
+export const getReviews = (category, query) => {
+  if (query) {
+    if (category) {
+      return ncGamesApi.get(`/reviews?category=${category}&sort_by=${query}`);
+    } else {
+      return ncGamesApi.get(
+        `/reviews?sort_by=${query}&sort_by=title&order=asc`
+      );
+    }
```
Comment (popatre) [inline src/api.js]: This is a bit of a clunky way to do it. You have a params object as part of axios which will do all this for you.

https://masteringjs.io/tutorials/axios/get-query-params

**Review by popatre — CHANGES_REQUESTED** [review]
It still doesnt quite work. 

If you change the order to, then refresh the page, you dont get the same results, which you would expect. 
Also you're missing sort_by in the url
`hidden-roles?owner&order=desc`
should be `hidden-roles?sort_by=owner&order=desc`

I would recommend either:

- Carry on using searchparams, but make sure you use the drop down to change the params and then get the value from there (not setting in state)

- Not using search params, and just keep it as it is. You currently save them in state, and change the url. You're not actually using the url, so it currently makes it not needed. 
The functionality works, it just doesnt match the params if you go to it manually.

**Pull request merged**